src/middleware/router.coffee

```diff
@@ -134,13 +139,25 @@ sendRequestToRoutes = (ctx, routes, next) ->
           # on failure
           handleServerError ctx, reason
       else
-        promise = buildNonPrimarySendRequestPromise ctx, route, options, path
+        logger.info "executing non primary: #{route.name}"
+        nonPrimaryPromise = buildNonPrimarySendRequestPromise ctx, route, options, path
+
+
 
       promises.push promise
+      nonPrimaryPromises.push nonPrimaryPromise
 
     (Q.all promises).then ->
       next()
 
+    (Q.all nonPrimaryPromises).then ->
+      messageStore.storeResponse ctx, ->
+        logger.info 'Storing non primary route responses'
+        stats.incrementTransactionCount ctx, ->
+        stats.measureTransactionDuration ctx, ->
```

Comment:
We may also want to split out the stats function into primary and nonPrimary functions, similar to above.